src/partition/InputLines.cc

```diff
@@ -199,6 +230,8 @@ class InputLines::Impl {
         boost::lock_guard<boost::mutex> lock(_mutex);
         return _blockCount == 0;
     }
+  
+    std::vector<std::string> getParameterNames() const { return _paramNames; }
```

Comment:
(for efficiency) Redeclare the return type as:
```
std::vector<std::string> const& getParameterNames() const { return _paramNames; }
```

Reply:
This function is not needed anymore